sdk/nodejs/automation/stack.ts

```diff
@@ -607,14 +607,21 @@ Event: ${line}\n${e.toString()}`);
 
         let logPromise: Promise<ReadlineResult> | undefined;
         let logFile: string | undefined;
+        let loggedSummary: OpMap | undefined;
         // Set up event log tailing
-        if (opts?.onEvent) {
+        if (opts?.previewOnly || opts?.onEvent) {
             const onEvent = opts.onEvent;
             logFile = createLogFile("destroy");
             args.push("--event-log", logFile);
 
             logPromise = this.readLines(logFile, (event) => {
-                onEvent(event);
+                if (opts?.previewOnly && event.summaryEvent) {
+                    loggedSummary = event.summaryEvent.resourceChanges;
```

Comment:
Same comment as in the refresh PR (do we want to grab the whole summary event), but otherwise, nice!